### PR TITLE
chore: add bug report form

### PR DESCRIPTION
**Pull request by clason**
Replace the Markdown issue template (which just tends to get deleted...) with a Github Form, which 
1. makes it much clearer what should be filled in,
2. removes the explanations on the submitted issue, and
3. makes most fields mandatory :)

I have added a minimal `init.lua` for reproducing steps which has the advantage over the old `init.vim` that it explicitly sets `runtimepath` and `packpath` to something clean so that user plugins/packages do not interfere. Two things to note here:

1. I have added `fzf-native` by default (which I think it should be anyway) so we can instantly weed out "it's slow" issues ;)
2. Bootstrapping `plenary` takes a significant time (first time only, though) -- hence the "progress output" -- but not sure what can be done (by Plenary) about this...

### CODE REVIEW

**Review by Conni2461 — APPROVED** [review]
This looks awesome. Thanks you very much :)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+             },
+             config = {
+               package_root = package_root,
+               compile_path = install_path .. '/plugin/packer_compiled.lua',
+               display = { non_interactive = true },
+             },
+           }
+         end
+         _G.load_config = function()
+           require('telescope').setup()
```
[the remark this answers is not in the document]
Comment (clason) [reply .github/ISSUE_TEMPLATE/bug_report.yml:76]: My understanding is that it overrides (only) the file sorter by default?

**Comment by tami5** [comment]
😍 😍 😍 😍 😍  well done @clason

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (clason) [reply .github/ISSUE_TEMPLATE/bug_report.yml:76]: (and that's what looks like happened in my tests :])

[the remark this answers is not in the document]
Comment (clason) [reply .github/ISSUE_TEMPLATE/bug_report.yml:76]: Always good to double-check :)

**Comment by fdschmidt93** [comment]
Looks amazing :) as a note to ourselves: #1066 should be added once merged

One thing I'm wondering about. The template rightfully implements required checks to report bugs though can discourage  support questions (that may or may not be bugs). I guess the feature request template could still be (ab-)used for this. Should we address this more explicitly somehow, say point to gitter for help? Maybe overthinking this though :sweat_smile:

**Review by fdschmidt93 — APPROVED** [review]

**Comment by clason** [comment]
> One thing I'm wondering about. The template rightfully implements required checks to report bugs though can discourage support questions (that may or may not be bugs). I guess the feature request template could still be (ab-)used for this. Should we address this more explicitly somehow, say point to gitter for help? Maybe overthinking this though 😅

Good question. My stance is that bug reports are bug reports and should not be abused for support requests. Whether these are welcome as Github issues is something you need to decide (and enforce); you can then add either
1. a free-form "support" template (which could still have required fields, for example related to topics) or
2. a link to Gitter 
on the "New issue" page.

(If you're going with route #2, I can easily add this to this PR.)

**Comment by clason** [comment]
I add (in a separate commit) both blank issue templates and a link to Gitter -- let me know which one you'd like to keep. 

(If you do keep the blank issue template, I'd recommend being _very_ strict about insta-closing bug reports that side-step the form...)

**Comment by fdschmidt93** [comment]
I'd say let's point to gitter and maybe we can add a FAQ in #1024. What's outstanding and onto us is docs for cool but complex stuff (eg see #1042) to address support. For most support, we could also advise people to use telescope to browse help_tags (which have great coverage); sounds maybe cynical though it would probably help a lot :laughing: that is something (in nice words :) we should probably add to "before reporting" though I suppose.

**Comment by clason** [comment]
oh, and I should make absolutely sure everyone is aware that I'm just ripping off @dundargoc's hard work on the main neovim repo here!

**Comment by clason** [comment]
OK, I'm going ahead and merging this since a) nobody complained and b) I want to see how it looks live. We can iterate further if anyone has additional suggestions.

(And fair warning: I have zero qualms about committing straight to master when tinkering with those `.yml` files.)

**Comment by joelpalmer** [comment]
These are wonderfully helpful for those submitting and triaging bugs. Great addition to the repo! The only bummer is no preview when building the forms. 😄  Ha! Very nice!

**Comment by clason** [comment]
Yes, that's a known problem (with forms): there's no preview for code (only markdown) textareas. I hope that will get fixed at some point, although I wouldn't hold my breath...

**Pull request merged**